### PR TITLE
Add Kinesis logging support

### LINKED ISSUE #350: Feature Request - logging_kinesis
Although currently only limited availability, Fastly support AWS [Kinesis Data Streams](https://docs.fastly.com/en/guides/log-streaming-amazon-kinesis-data-streams) for logging.

We are keen to test this as soon as possible so it would be exceptionally useful if you could add a corresponding `logging_kinesis` argument to the `fastly_service_v1` resource ahead of general release.

### PR DESCRIPTION
**Pull request by MerlinDMC**
## Proposed Change(s)

- Updates `go-fastly` to version `v2.1.0` to support Kinesis.
- Adds support for provisioning Kinesis logging endpoints.
- Updates `fastly_service_v1` and `fastly_service_compute` docs to reflect the change.

## Relevant Link(s) / Doc(s)

- [API Docs](https://developer.fastly.com/reference/api/logging/kinesis/)
- [Relevant Fastly API Go client library file](https://github.com/fastly/go-fastly/blob/master/fastly/kinesis.go)

## Validation

```shell
export FASTLY_API_KEY="foo"; export TF_ACC=1; make testacc TESTARGS='-run=[kK]inesis'
```

fixes #350

### CODE REVIEW

File: fastly/resource_fastly_service_compute.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ var computeService = &BaseServiceDefinition{
 		NewServiceLoggingOpenstack(computeAttributes),
 		NewServiceLoggingDigitalOcean(computeAttributes),
 		NewServiceLoggingCloudfiles(computeAttributes),
+		NewServiceLoggingKinesis(vclAttributes),
```
Comment (Integralist) [inline fastly/resource_fastly_service_compute.go]:
```suggestion
		NewServiceLoggingKinesis(computeAttributes),
```

**Review by Integralist — CHANGES_REQUESTED** [review]
👋🏻 I ran your branch/tests locally and found they were failing so I noticed this issue in the code that when updated would cause the tests to pass.

I'll give another proper/full review tomorrow.

Thanks.

File: website/docs/r/service_compute.html.markdown [hunk 2]
```diff
@@ -522,6 +524,17 @@ The `logging_cloudfiles` block supports:
 * `timestamp_format` - (Optional) The strftime specified timestamp formatting (default `%Y-%m-%dT%H:%M:%S.000`).
 
 
+### logging_kinesis block
+
+The `logging_kinesis` block supports:
+
+* `name` - (Required) The unique name of the Logshuttle logging endpoint.
```
Comment (Integralist) [inline website/docs/r/service_compute.html.markdown]:
```suggestion
* `name` - (Required) The unique name of the Kinesis logging endpoint.
```

File: website/docs/r/service_v1.html.markdown [hunk 3]
```diff
@@ -888,6 +890,20 @@ The `logging_cloudfiles` block supports:
 * `placement` - (Optional) Where in the generated VCL the logging call should be placed. Can be `none` or `waf_debug`.
 * `response_condition` - (Optional) The name of an existing condition in the configured endpoint, or leave blank to always execute.
 
+### logging_kinesis block
+
+The `logging_kinesis` block supports:
+
+* `name` - (Required) The unique name of the Logshuttle logging endpoint.
```
Comment (Integralist) [inline website/docs/r/service_v1.html.markdown]:
```suggestion
* `name` - (Required) The unique name of the Kinesis logging endpoint.
```

File: website_src/docs/r/arguments/logging_kinesis.markdown.tmpl [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{{define "logging_kinesis"}}* `logging_kinesis` - (Optional) A Kinesis endpoint to send streaming logs to.
+[Defined below](#logging_kinesis-block).{{end}}
+
+{{define "logging_kinesis_block"}}
+### logging_kinesis block
+
+The `logging_kinesis` block supports:
+
+* `name` - (Required) The unique name of the Logshuttle logging endpoint.
```
Comment (Integralist) [inline website_src/docs/r/arguments/logging_kinesis.markdown.tmpl]:
```suggestion
* `name` - (Required) The unique name of the Kinesis logging endpoint.
```

**Review by Integralist — COMMENTED** [review]
👋🏻 I noticed just a couple of minor corrections.

**Comment by MerlinDMC** [comment]
Copy and Paste 😞

**Review by Integralist — APPROVED** [review]
Awesome @MerlinDMC thank you for taking the time to put this PR together! 

This looks good to me.

I'm now running the full test suite on your branch before I merge, but unfortunately it can take many hours to complete (we're working on fixing that btw).

**Comment by MerlinDMC** [comment]
👍 Thanks for taking the time to review and test it. We're looking forward to test the logging stream via Kinesis 😉

**Pull request merged**